src/refdb_fs.c

```diff
@@ -1064,6 +1064,30 @@ static int packed_write(refdb_fs_backend *backend)
 	return error;
 }
 
+static int packed_delete(refdb_fs_backend *backend, const char *ref_name)
+{
+	size_t pack_pos;
+	int error;
+
+	if ((error = packed_reload(backend)) < 0)
+		goto cleanup;
+
+	/* If a packed reference exists, remove it from the packfile and repack */
+	if ((error = git_sortedcache_wlock(backend->refcache)) < 0)
+		goto cleanup;
+
+	if (!(error = git_sortedcache_lookup_index(
+			&pack_pos, backend->refcache, ref_name)))
+		error = git_sortedcache_remove(backend->refcache, pack_pos);
+
+	git_sortedcache_wunlock(backend->refcache);
+
+	error = packed_write(backend);
```

Comment:
This line is clobbering the previous error code. Furthermore, I'd expect no `packed_write` to be necessary in case we do not remove the entry. Maybe you can just move the unlock into `cleanup` and set a flag as soon as the lock has been taken. Then you can easily convert the lookup/remove into our usual `if ((error = fn(...)) < 0)) goto cleanup;`.

Reply:
I don't think I can move the unlock to `cleanup` because the `backend->refcache` lock is not reentrant and `packed_write` will (dead-)lock it. I'll fix the clobber though.